### ISSUE TITLE
Incorrect translation of timing phrases in the presence of certain choice types

### ISSUE THREAD

**Issue by brynrhodes** [cqframework/clinical_quality_language#331]
In cases where timing phrases allow both point and interval types on either side, if one of the inputs is a choice type with both point and interval types, the timing phrase will assume the point-valued choice (because the logic sees it as "not an interval or list type".

This should, at a minimum, be corrected to reflect the intent of the timing phrase (interval-valued for phrases like included in and during).

An alternative solution is to preserve the choice, adding translations for both options.

    define Procedures:
      [Procedure] P
        where P.performed included in "Measurement Period"

This translates as an "in" because the timing phrase is looking for whether the expression is not list- or interval-valued. Because performed is a "choice", it could be interval-valued, or it could be datetime-valued. In that case, it should continue to treat it as an interval timing phrase not a left-point-valued timing phrase. An alternative of course is to translate both choices:

    ((P.performed as DateTime) in "Measurement Period" or (P.performed as Interval<DateTime>) included in "Measurement Period")

**Comment by cmoesel** [comment]
I don't think this translation would work:
```
((P.performed as DateTime) in "Measurement Period" or (P.performed as Interval<DateTime>) included in "Measurement Period")
```

* One of `(P.performed as DateTime)` or `(P.performed as Interval<DateTime>)` will always be `null` (since it can't be both at the same time).
* `In` and `Included In` for intervals both specify that if an argument is `null`, the result is `null`.
* This means one of the `or` arguments will always be `null`.
* This means that the overall expression can _never_ return `false`.  It is only capable of returning `true` or `null`.

I guess you would need to do this instead?
```
(((P.performed as DateTime) in "Measurement Period") is true or ((P.performed as Interval<DateTime>) included in "Measurement Period") is true)
```

**Comment by brynrhodes** [comment]
@cmoesel that's a great point, thank you. That additional complexity makes me lean even more strongly towards the first option, assuming the interval-valued timing phrase.

**Comment by cmoesel** [comment]
Right.  And to be sure I understand the first option -- this works because many interval operations are overloaded to allow the LHS to be a point, right?

**Comment by brynrhodes** [comment]
Yes, the timing phrase allows mixtures of point and interval arguments for many of the interval operations.

**Comment by brynrhodes** [comment]
I ended up addressing this by adding support for interval promotion and demotion and removing the special case translations. This provided a much cleaner implementation.

Included in the [1.3 CQL Ballot](https://github.com/cqframework/clinical_quality_language/releases/tag/v1.3.0-SNAPSHOT) release.
